src/test/java/cucumber/definitionSteps/StepDefinitions.java

```diff
@@ -0,0 +1,163 @@
+package cucumber.definitionSteps;
+
+import cucumber.setUp.CucumberContext;
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.openqa.selenium.Alert;
+import org.testng.Assert;
+import pages.ActionsPage;
+import pages.HomePage;
+import pages.LoginPortalPage;
+import pages.OptionsPage;
+
+import java.util.List;
+
+public class StepDefinitions {
+  private final CucumberContext context;
+  private ActionsPage actionsPage;
+  private OptionsPage optionsPage;
+  private HomePage homePage;
+  private LoginPortalPage loginPortalPage;
+  private String getAlertMessage;
+
+  public StepDefinitions(CucumberContext cucumberContext) {
+    this.context = cucumberContext;
+  }
+
+  @Before
+  public void setUp() {
+    context.setUp();
+    homePage = context.homePage;
+    loginPortalPage = context.pageFactoryManager.getPage(LoginPortalPage.class);
+    actionsPage = context.pageFactoryManager.getPage(ActionsPage.class);
+    optionsPage = context.pageFactoryManager.getPage(OptionsPage.class);
+  }
+
+  @After
+  public void tearDown() {
+    context.driver.quit();
+  }
+
+  @Given("User opens Home page")
+  public void openHomePage() {
+    homePage.openHomePage(context.baseUrl);
+  }
+
+  @When("User clicks on the login portal link")
+  public void clickOnTheLoginPortalLink() {
+    homePage.clickOnLoginPortalLink();
+  }
+
+  @And("User switches to another tab")
+  public void switchToAnotherTab() {
+    List<String> tabsList = homePage.tabsList();
+    homePage.switchToTab(tabsList.get(context.FIRST));
+  }
+
+  @And("^User logs in with name - \"([A-Za-z]+)\", password - \"(.{8,})\"$")
```

Comment:
For me, it's more convenient to use just {string} instead of regex. It improves readability. But it's up to you :)